tests/python_package_test/test_engine.py

```diff
@@ -3578,3 +3578,26 @@ def test_boost_from_average_with_single_leaf_trees():
     preds = model.predict(X)
     mean_preds = np.mean(preds)
     assert y.min() <= mean_preds <= y.max()
+
+
+@pytest.mark.parametrize('verbosity', [-1, 0])
+def test_verbosity_can_suppress_alias_warnings(capsys, verbosity):
+    X, y = make_synthetic_regression()
+    ds = lgb.Dataset(X, y)
+    params = {
+        'num_leaves': 3,
+        'subsample': 0.75,
+        'bagging_fraction': 0.8,
+        'force_col_wise': True,
+        'verbosity': verbosity,
+    }
+    lgb.train(params, ds, num_boost_round=1)
+    expected_msg = (
+        '[LightGBM] [Warning] bagging_fraction is set=0.8, subsample=0.75 will be ignored. '
+        'Current value: bagging_fraction=0.8'
+    )
+    stdout = capsys.readouterr().out
+    if verbosity == -1:
+        assert stdout == ''
```

Comment:
Instead of testing that `stdout` is empty, I think this should be changed to checking that there are 0 LightGBM-generated messages (i.e. those containing `[LightGBM]`) emitted.

That would prevent the case where this test fails when one of `lightgbm`'s dependencies emits some log messages (like a deprecation warning or something) that `lightgbm` does not try to suppress based on the value of its `verbosity` setting.

I proposed something similar in the R package's tests when working on #4862, e.g. https://github.com/microsoft/LightGBM/blob/fb37e5074ad0163afd9d88366b58075899775cfa/R-package/tests/testthat/test_basic.R#L2312-L2315